cardstack/src/services/historical-pricing-service.ts

```diff
@@ -1,51 +1,63 @@
-import { cryptoCurrencies } from '@cardstack/cardpay-sdk';
-import { CRYPTOCOMPARE_API_KEY } from 'react-native-dotenv';
+import { cryptoCurrencies, NativeCurrency } from '@cardstack/cardpay-sdk';
+import { format } from 'date-fns';
 
+import { getExchangeRatesQuery } from '@cardstack/services/hub/hub-service';
 import { removeCPXDTokenSuffix } from '@cardstack/utils';
 
 import logger from 'logger';
 
-const CryptoCompareAPIBaseURL = `https://min-api.cryptocompare.com/data/pricehistorical?&api_key=${CRYPTOCOMPARE_API_KEY}`;
-
 export const fetchHistoricalPrice = async (
   symbol: string,
-  timestamp: string | number,
-  nativeCurrency: string
+  timestamp: number,
+  nativeCurrency: NativeCurrency | string
 ): Promise<number> => {
   try {
     if (!symbol) {
       return 0;
     }
 
+    // date needs to be formatted as yyyy-MM-dd
+    const formattedDate = format(timestamp, 'yyyy-MM-dd');
+
     // For cpxd tokens we chop off .CPXD in order to get mainnet's historical price
     const tokenSymbol = removeCPXDTokenSuffix(symbol);
 
     // cryptocompare does not support CARD price history correctly,
     // so uses kucoin exchange to get CARD price in USDT(kucoin supports only USDT) and convert it to native currency
     if (tokenSymbol === cryptoCurrencies.CARD.currency) {
-      const usdtPriceData = await (
-        await fetch(
-          `${CryptoCompareAPIBaseURL}&fsym=${tokenSymbol}&tsyms=USDT&ts=${timestamp}&e=kucoin`
-        )
-      ).json();
+      const { data: kucoinRate } = await getExchangeRatesQuery({
+        from: tokenSymbol,
+        to: 'USDT',
+        date: formattedDate,
+        e: 'kucoin',
+      });
 
-      const usdtPrice = usdtPriceData.CARD.USDT;
+      const usdtPrice = kucoinRate?.USDT;
 
-      const nativeCurrencyPriceForUSDT = await fetchHistoricalPrice(
-        'USDT',
-        timestamp,
-        nativeCurrency
-      );
+      const { data: USDTRate } = await getExchangeRatesQuery({
+        from: 'USDT',
+        to: tokenSymbol,
+        date: formattedDate,
+      });
 
-      return usdtPrice * nativeCurrencyPriceForUSDT;
+      const nativeCurrencyPriceForUSDT = USDTRate?.[tokenSymbol];
+
+      if (usdtPrice && nativeCurrencyPriceForUSDT) {
+        return usdtPrice * nativeCurrencyPriceForUSDT;
+      } else {
+        return 0;
+      }
```

Comment:
superNit: we could default usdtPrice and nativeCurrencyPriceForUSDT to 0, and simply return the multiplication

Reply:
The problem here is that both ustdPrice and nativeCurrencyPriceForUSDT are promises, so they both can be undefined.... what do you think if I switch to:

```suggestion
      if (!usdtPrice || !nativeCurrencyPriceForUSDT) {
        return 0;
      }

      return usdtPrice * nativeCurrencyPriceForUSDT;
```